File: ambeth/Ambeth.IoC/ambeth/accessor/IAccessorTypeProvider.cs
using De.Osthus.Ambeth.Typeinfo;
using System;

namespace De.Osthus.Ambeth.Accessor
{
    public interface IAccessorTypeProvider
    {
        AbstractAccessor GetAccessorType(Type type, IPropertyInfo property);

	    V GetConstructorType<V>(Type targetType);
    }
}

File: jambeth/jambeth-ioc/src/main/java/de/osthus/ambeth/accessor/IAccessorTypeProvider.java
package de.osthus.ambeth.accessor;

import de.osthus.ambeth.typeinfo.IPropertyInfo;

public interface IAccessorTypeProvider
{
	AbstractAccessor getAccessorType(Class<?> type, IPropertyInfo property);

	<V> V getConstructorType(Class<V> delegateType, Class<?> targetType);
}
